.github/workflows/lint.yml

```diff
@@ -0,0 +1,62 @@
+name: Lint
+
+on:  [push, pull_request]
+
+jobs:
+  build:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    permissions:
```

Comment:
Just for my knowledge, what are these setups?

Reply:
just granting minimal permissions for workflow but I removed as we don't actually need this much at this stage.